snmp.go

```diff
@@ -45,7 +44,27 @@ type snmpPorterrorsMeasure struct {
 	ifInUnknownProtos uint
 }
 
-func (fm *Frontman) runSNMPCheck(check *SNMPCheck) (map[string]interface{}, error) {
+func (check SNMPCheck) uniqueID() string {
+	return check.UUID
+}
+
+func (check SNMPCheck) run(fm *Frontman) (*Result, error) {
+
+	res := &Result{
```

Comment:
all lines are repeated from above except for one type:


      res := &Result{
		Node:      fm.Config.NodeName,
		CheckType: "serviceCheck",
		CheckUUID: check.UUID,
		Check:     check.Check,
		Timestamp: time.Now().Unix(),
	}
	if check.UUID == "" {
		return res, fmt.Errorf("missing checkUuid key")
	}
	if check.Check.Connect == "" {
		return res, fmt.Errorf("missing data.connect key")
	}


a reusable function could help here

Reply:
It's due to them being different types.

Reply:
but building logic is shared no? a function could resolve this